docs/admin/additional_features/oidc.rst

```diff
@@ -41,13 +41,26 @@ JWKS_URI                       https://your-oidc-provider.com/certs
 
 The following additional configuration options are available:
 
-============================== =================================================== ========================
-Value                          Usage                                               Default value
-============================== =================================================== ========================
-scopes                         Scopes to request from the RP (for additional data) ``openid profile email``
-end_session_endpoint           redirect the user to the logout page of the IDP     None (no redirect)
-default groups                 groups to add all users logging in with this IDP to None (no groups)
-============================== =================================================== ========================
+============================== ==================================================== ========================
+Value                          Usage                                                Default value
+============================== ==================================================== ========================
+scopes                         Scopes to request from the IDP (for additional data) ``openid profile email``
+end_session_endpoint           redirect the user to the logout page of the IDP      None (no redirect)
+============================== ==================================================== ========================
+
+Apart from the values above, you can also configure the handling of groups for each identity provider.
+If your identity providers supports exposing groups in the identity token, ephios can use this information to automatically assign users to groups.
+You may be required to configure your identity provider to expose the groups in the identity token, please refer to the documentation of your identity provider for more information.
+It may also be neccessary to request an additional scope from the identity provider, e.g. ``groups``. Append this scope to the ``scopes`` setting if required.
+To enable the functionality, you need to provide the name of the claim that contains the group information.
+For example, if your identity provider exposes the groups in a claim called ``groups``, you would enter ``groups`` there.
+You can also access nested claims by using a dot-separated path, e.g. ``extra_information.groups`` if the group list is situated inside an ``extra_information`` dict in the identity token.
+The value of the claim must be a list of strings, where each string is the name of a group in ephios (case-insensitive).
+The user will be a member of all groups that are listed in the claim, meaning they will be removed from groups that they have been manually assigned to in ephios and that are not listed in the claim.
```

Comment:
I don't know whether removing users from manually assigned groups is the best way to do it. Using groups in ephios that do not exist in the more abstract IDP.

Couldn't we still have a multi-select for groups managed by the OIDC, and when the groups claim is configured, the user is not added to all, but to the oidc-managed groups provided by the claim.

Reply:
so two different multiselects on the provider? I think that is really confusing for people configuring it. And that would make the create_missing_groups impossible if only predefined groups would be respected

Reply:
we decided against that